packages/grid/src/GridModel.ts

```diff
@@ -183,6 +188,45 @@ abstract class GridModel<
   ): IColumnHeaderGroup | undefined {
     return undefined;
   }
+
+  /**
+   * Gets the tokens in the cell at column and row, based on the visible text
+   * @param column The model column
+   * @param row The model row
+   * @param visibleLength The length of the visible text
+   * @returns An array of Tokens in the cell
+   */
+  tokensForCell(
+    column: ModelIndex,
+    row: ModelIndex,
+    visibleLength: number
+  ): Token[] {
+    return this.getCachedTokensForCell(column, row, visibleLength);
+  }
+
+  getCachedTokensForCell = memoizeClear(
+    (column: ModelIndex, row: ModelIndex, visibleLength: number): Token[] => {
```

Comment:
We're memoizing on the wrong thing here. Text can change for a given `column`/`row` combination, so you might get stale results. Should instead memoize on the text. Text could also be the same in multiple cells, and then they'd have the same tokens.